packages/integration-tests/suites/replay/privacyBlock/test.ts

```diff
@@ -0,0 +1,31 @@
+import { expect } from '@playwright/test';
+import { EventType } from '@sentry-internal/rrweb';
+import type { RecordingEvent } from '@sentry/replay/build/npm/types/types';
+
+import { sentryTest } from '../../../utils/fixtures';
+import { envelopeRequestParser } from '../../../utils/helpers';
+import { shouldSkipReplayTest, waitForReplayRequest } from '../../../utils/replayHelpers';
+
+sentryTest('should allow to manually block elements', async ({ getLocalTestPath, page }) => {
+  if (shouldSkipReplayTest()) {
+    sentryTest.skip();
+  }
+
+  const reqPromise0 = waitForReplayRequest(page, 0);
+
+  await page.route('https://dsn.ingest.sentry.io/**/*', route => {
+    return route.fulfill({
+      status: 200,
+      contentType: 'application/json',
+      body: JSON.stringify({ id: 'test-id' }),
+    });
+  });
+
+  const url = await getLocalTestPath({ testDir: __dirname });
+
+  await page.goto(url);
+  const replayPayload = envelopeRequestParser<RecordingEvent[]>(await reqPromise0, 5);
+  const checkoutEvent = replayPayload.find(({ type }) => type === EventType.FullSnapshot);
+
+  expect(JSON.stringify(checkoutEvent?.data, null, 2)).toMatchSnapshot('privacy.json');
```

Comment:
m/h: Two things here:

* `replayHelpers.ts` has two functions, we can use to obtain the snapshot without having to manually call the envelope parser. This will also ensure that getting the snapshot works if we enabled compression in `init.js`. Let's either use `getFullRecordingSnapshot` or `getReplayRecordingContent` here.
* Let's use the `normalize` helper function I added to stringify the snapshot. I suggest we update the normalize function to treat the pixel/size values as they're apparently flakey. Maybe, it's enough to round them to the closes integer or we might even need to hard-redact them (like the time offset in incremental snapshots). 

I've been meaning to go over the tests we added before we merged all these helpers and adjust them but didn't get to it yet. Would be great if we could adjust these two tests.